File: Guerreiro.java
class Guerreiro extends Personagem{
    protected  int defesa = 5;

    public Guerreiro(String nome, int vida, int nivel, int defesa){
        super(nome, vida, nivel);
        this.defesa = defesa;
    }

    //Getters
    public int getdefesa(){
        return(defesa);
    }

    //Setters
    public void setdefesa(int defesa){
        if (defesa >= 0){
            this.defesa = defesa;
        } else{
            System.out.printf("%nInsira um valor válido para a defesa%n");
        }
    }
    
    @Override
    public int receberDano(int dano){
        return super.receberDano(Math.max(0, dano - defesa));
    }

    @Override
    public void ficha(){
        super.ficha();
        System.out.printf("%nDefesa: %d%n", defesa);
    }
    
}

File: Herança-Subclasses-de-Personagem/Guerreiro.py
from Personagem import Personagem


class Guerreiro(Personagem):

    def __init__(self, nome, vida, nivel, defesa):
        super().__init__(nome, vida, nivel)
        self.defesa = defesa

    # Getter
    def getdefesa(self):
        return self.defesa

    # Setter
    def setdefesa(self, defesa):
        if defesa >= 0:
            self.defesa = defesa
        else:
            print("\nInsira um valor válido para a defesa")

    # Override
    def receberDano(self, dano):
        return super().receberDano(max(0, dano - self.defesa))

    # Override
    def ficha(self):
        super().ficha()
        print(f"\nDefesa: {self.defesa}")
